code/game/objects/items/stacks/stack.dm

```diff
@@ -22,8 +22,9 @@
 	return
 
 /obj/item/stack/Destroy()
-	if (src && usr && usr.machine==src)
+	if (usr && usr.machine==src)
 		usr << browse(null, "window=stack")
+	src.loc = null
```

Comment:
Not sure why you'd want to null out the loc here, considering it will get nulled out higher up the Destroy chain.